cpp/tests/test_abm.cpp

```diff
@@ -238,63 +246,173 @@ TEST(TestLocation, beginStep)
-                                      vaccination_state);
-    home.add_person(infected3);
-    infected3.migrate_to(home, location1, {1});
+TEST(TestLocation, reachCapacity)
+{
+    using testing::Return;
 
-    //cache precomputed results
-    auto dt = mio::abm::seconds(8640);
-    location1.begin_step(dt, params);
+    auto world     = mio::abm::World();
+    auto home_id   = world.add_location(mio::abm::LocationType::Home);
+    auto school_id = world.add_location(mio::abm::LocationType::School);
+
+    ScopedMockDistribution<testing::StrictMock<MockDistribution<mio::UniformDistribution<double>>>> mock_uniform_dist;
+    EXPECT_CALL(mock_uniform_dist.get_mock(), invoke)
+        .Times(testing::AtLeast(10))
+        .WillOnce(testing::Return(0.8)) // draw vaccination state
+        .WillOnce(testing::Return(0.8)) // draw random work group
+        .WillOnce(testing::Return(0.8)) // draw random school group
+        .WillOnce(testing::Return(0.8)) // draw random work hour
+        .WillOnce(testing::Return(0.8)) // draw random school hour
+        .WillOnce(testing::Return(0.8)) // draw vaccination state
+        .WillOnce(testing::Return(0.8)) // draw random work group
+        .WillOnce(testing::Return(0.8)) // draw random school group
+        .WillOnce(testing::Return(0.8)) // draw random work hour
+        .WillOnce(testing::Return(0.8)); // draw random school hour
+    // .WillRepeatedly(testing::Return(1.0));
+
+    auto& p1 = world.add_person(home_id, mio::abm::InfectionState::Carrier, mio::abm::AgeGroup::Age5to14);
+    auto& p2 = world.add_person(home_id, mio::abm::InfectionState::Susceptible, mio::abm::AgeGroup::Age5to14);
+
+    p1.set_assigned_location(school_id);
+    p2.set_assigned_location(school_id);
+    p1.set_assigned_location(home_id);
+    p2.set_assigned_location(home_id);
+
+    auto& home   = world.get_individualized_location(home_id);
+    auto& school = world.get_individualized_location(school_id);
+    school.set_capacity(1, 66);
+
+    ScopedMockDistribution<testing::StrictMock<MockDistribution<mio::ExponentialDistribution<double>>>>
+        mock_exponential_dist;
+    EXPECT_CALL(mock_exponential_dist.get_mock(), invoke).WillRepeatedly(Return(1.)); //no state transitions
+
+    world.evolve(mio::abm::TimePoint(0) + mio::abm::hours(8), mio::abm::hours(1));
 
-    ASSERT_TRUE(std::abs(location1.get_cells()[0].cached_exposure_rate[{age, vaccination_state}] - 0.9) < 0.001);
-    ASSERT_TRUE(std::abs(location1.get_cells()[1].cached_exposure_rate[{age, vaccination_state}] - 1) < 0.001);
-    ASSERT_TRUE(std::abs(location1.get_cells()[2].cached_exposure_rate[{age, vaccination_state}]) < 0.001);
+    ASSERT_EQ(p1.get_location_id().type, mio::abm::LocationType::School);
+    ASSERT_EQ(p2.get_location_id().type, mio::abm::LocationType::Home); // p2 should not be able to enter the school
+    ASSERT_EQ(school.get_subpopulations().sum(), 1);
+    ASSERT_EQ(home.get_subpopulations().sum(), 1);
```

Comment:
Tests look good. There should be a test for the function compute relative_transmission_risk. It's a perfect function for a unit-test. ;-)

Reply:
added a test for compute_relative_transmission_risk